cirrus/server/cirrus/main.py

```diff
@@ -109,6 +110,41 @@ def initialize_glean():
     return pings, metrics
 
 
+def get_metrics_data(enrolled_partial_configuration: Dict[str, Any]):
+    enrollments = enrolled_partial_configuration.get("enrollments", [])
+    experiment_slug = enrollments[0].get("slug") if enrollments else ""
+    branch_slug = (
+        enrollments[0].get("status", {}).get("Enrolled", {}).get("branch")
+        if enrollments
+        else ""
+    )
+    events = enrollment_id = enrolled_partial_configuration.get("events", [])
+    enrollment_id = events[0].get("enrollment_id") if events else ""
+    experiment_type = (
+        app.state.remote_setting.get_recipe_type(experiment_slug)
+        if experiment_slug
+        else ""
+    )
+    return experiment_slug, branch_slug, enrollment_id, experiment_type
```

Comment:
To collate the enrollment data into the data that should be sent through Glean you'll actually want it to iterate across all the enrollments, filter them down to only those with the `Enrolled` status, and return a list of enrollment extra dicts (or a specific type if you'd prefer).

```suggestion
def get_metrics_data(enrolled_partial_configuration: Dict[str, Any]):
    result = []
    enrollments = enrolled_partial_configuration.get("enrollments", [])
    
    for enrollment in enrollments:
        # Check if status is `Enrolled`
        # Find enrollment event in `events` to get enrollment uuid
        # Put together dict/object with values noted above
        # Append dict to `result`
        pass
        
    return result
```